app/controllers/api/v2/ext_app/containers_controller.rb

```diff
@@ -50,6 +50,15 @@ def rebootstrap
     render json: ::Api::V2::ExtApp::ContainerSerializer.new(@container).to_h
   end
 
+  # POST /:hostname/update
```

Comment:
why not PATCH/PUT ?

Reply:
also who will call this API?

Reply:
> why not PATCH/PUT?

cc @galuhest.

> also who will call this API?

Previously we don't have **bootstrapper** attribute on Pathfinder. BaritoMarket bootstraps the containers so the **bootstrapper** is stored on BaritoMarket. Currently, **bootstrapper** attributes for existing containers in Pathfinder are set to `[]`.

How the deployment re-bootstrap a container is by checking **attributes in the container** and **attributes generated by the deployment**. If the attributes are different, **attributes generated by the deployment** will be set to **attributes in the container** and the container will be bootstrapped after that.

If we don't synchronize **bootstrapper** attribute from BaritoMarket to Pathfinder, `[]` will be different with **attributes generated by the deployment**, so when we migrate, all of the existing containers in Barito will be bootstrapped.

We minimalize this by providing API to set the container attributes manually, so BaritoMarket can synchronize the **bootstrapper** attribute to Pathfinder. We tried our best to make sure that **attributes generated by the deployment** will be the same as *attributes stored in current BaritoMarket.** So, **attributes generated by the deployment** will be the same as synchronized **attributes in the container**, and the containers will not be re-bootstrapped while we migrate Barito to the new deployment model.

Reply:
it was a mistake mas @giosakti , i will change it to PUT

Reply:
thanks Galuh,

@nieltg ok got it,
can I assume that once we migrate to deployment model 100%, we don't need this anymore? since we only have to update attributes in deployment?

Reply:
For our use case, we think that we don't need it anymore.

Maybe for other people who use Barito, it'll be useful for them to migrate.

Reply:
cool, then this is something that can be deprecated in future release